platforms/software/maven/src/main/java/org/gradle/api/publish/maven/MavenPublication.java

```diff
@@ -263,11 +266,13 @@ public interface MavenPublication extends Publication {
      * Returns the complete set of artifacts for this publication.
      * @return the artifacts.
      */
+    @ToBeKeptEagerProperty(because = "Read-only nested like property")
     MavenArtifactSet getArtifacts();
```

Comment:
❗ Shouldn't need annotation as it's a `DOC`.

Reply:
Done